src/material/table/table-data-source.spec.ts

```diff
@@ -51,6 +52,19 @@ describe('MatTableDataSource', () => {
     it('should be able to correctly sort an array of strings and numbers', () => {
       testSortWithValues([3, 'apples', 'bananas', 'cherries', 'lemons', 'strawberries']);
     });
+
+    it('should unsubscribe from the re-render stream when disconnected', () => {
+      const spy = spyOn(dataSource._renderChangesSubscription!, 'unsubscribe');
+      dataSource.disconnect();
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should re-subscribe to the sort stream when re-connecting after being disconnected', () => {
```

Comment:
@andrewseguin I'm not totally sure whether re-connecting a disconnected data source is something we actually support. On one hand, the comment on `disconnect` says "Used by the MatTable. Called when it is destroyed. No-op.", but on the other, I can see a use case where the user would swap between data sources within the same table.

Reply:
Seems fine with me. Probably just remove the "No-op" part of the documentation though